ui/src/app/views/copy-notebook-modal/component.tsx

```diff
@@ -0,0 +1,185 @@
+        const errorMsg = response.status === 400 ?
+          'Notebook with the same name already exists in the targeted workspace.' :
+          'An error occurred while copying. Please try again.';
+
+        this.setState({
+          errorMsg: errorMsg,
+          state: State.ERROR,
+          loading: false
+        });
+      });
+  }
+
+  goToDestinationWorkspace() {
+    window.location.href = this.getWorkspaceUrl(this.state.destination);
+  }
+
+  getWorkspaceUrl(workspace: Workspace) {
+    return ['workspaces', workspace.namespace, workspace.id, 'notebooks'].join('/');
+  }
+
+  render() {
+    return (
+      <Modal onRequestClose={this.props.onClose}>
+        <ModalTitle>Copy to Workspace</ModalTitle>
+          { this.state.loading ?
+            <ModalBody style={{ textAlign: 'center' }}><Spinner /></ModalBody> :
+            <ModalBody>
+                {this.state.state === State.FORM && this.renderFormBody()}
+                {this.state.state === State.ERROR && this.renderErrorBody()}
+                {this.state.state === State.SUCCESS && this.renderSuccessBody()}
+            </ModalBody>
+        }
+        <ModalFooter>
+          <Button type='secondary' onClick={this.props.onClose}>
+              {this.getCloseButtonText()}
+          </Button>
+          {this.renderActionButton()}
+        </ModalFooter>
+      </Modal>
+    );
+  }
+
+  getCloseButtonText() {
+    if (this.state.state === State.FORM) {
+      return 'Cancel';
+    } else if (this.state.state === State.ERROR) {
+      return 'Close';
+    } else if (this.state.state === State.SUCCESS) {
+      return 'Stay Here';
+    }
+  }
+
+  renderActionButton() {
+    if (this.state.state === State.FORM) {
+      return (
+        <Button style={{ marginLeft: '0.5rem' }}
+          disabled={this.state.destination === null}
+          onClick={() => this.save()}>
+          Copy Notebook
+        </Button>
+      );
+    } else if (this.state.state === State.ERROR) {
+      return null;
+    } else if (this.state.state === State.SUCCESS) {
+      return (
+        <Button style={{ marginLeft: '0.5rem' }}
+          onClick={this.goToDestinationWorkspace}>
+          Go to Copied Notebook
+        </Button>
+      );
+    }
+  }
+
+  renderFormBody() {
+    return (
+      <div>
+        <div style={headerStyles.formLabel}>Destination *</div>
+        <Select
+            value=''
+            options={this.state.writeableWorkspaces.map(workspace => ({
+              'value': workspace,
+              'label': workspace.name
+            }))}
+            onChange={(value) => { this.setState({ destination: value }); }} />
+        <div style={headerStyles.formLabel}>Name *</div>
+        <TextInput
+            autoFocus
+            value={this.state.newName}
+            onChange={v => this.setState({ newName: v })}
+        />
+      </div>
+    );
+  }
+
+  renderErrorBody() {
+    return <div style={headerStyles.formLabel}>{this.state.errorMsg}</div>;
```

Comment:
I would have this error show up in the form, rather than as a separate modal body implementation. (Spin the button on loading, and if it fails, just have the error message show up on the body.)

Reply:
Within the form body? To allow the user to retry?

Reply:
Yes, within the form body.